server/src/main/java/org/opensearch/indices/NodeIndicesStats.java

```diff
@@ -251,7 +375,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
-    private Map<Index, CommonStats> createStatsByIndex() {
+    private Map<Index, CommonStats> createStatsByIndex(Map<Index, List<IndexShardStats>> statsByShard) {
         Map<Index, CommonStats> statsMap = new HashMap<>();
         for (Map.Entry<Index, List<IndexShardStats>> entry : statsByShard.entrySet()) {
             if (!statsMap.containsKey(entry.getKey())) {
```

Comment:
`if (!statsMap.containsKey(entry.getKey())) {` check looks unnecessary.

Reply:
Don't quite understand what you are referring to. A node can contain multiple indices and for every shard, we first check the index and if that index does not exists, we create a new object and insert else we add it to the existing object.